site/documentation/content/api/device-connection/index.md

```diff
@@ -107,6 +107,178 @@ Implementors of this API may return a *404* status code in order to indicate tha
+1. Otherwise, *all* adapter instances associated with any of the given gateway identifiers are returned.
+
+That means that for a device communicating via a gateway, the result is reduced to a *single element* list if an adapter instance for the device itself or its last known gateway is found. The adapter instance registered for the device itself is given precedence in order to ensure that a gateway having subscribed to commands for that *particular device* is chosen over a gateway that has subscribed to commands for *all* devices of a tenant.
+
+**Message Flow**
+
+
+{{< figure src="get_cmd_handling_adapter_instances.svg" title="Client retrieving the list of command-handling adapter instances for a device" alt="A client sends a request message for retrieving the list of command-handling adapter instances for a device and receives a response containing the information" >}}
+
+**Request Message Format**
+
+The following table provides an overview of the properties a client needs to set on a message to retrieve the command-handling adapter instances for a device in addition to the [Standard Request Properties]({{< relref "#standard-request-properties" >}}).
+
+| Name           | Mandatory | Location                 | AMQP Type | Description |
+| :------------- | :-------: | :----------------------- | :-------- | :---------- |
+| *subject*      | yes       | *properties*             | *string*  | MUST be set to `get-cmd-handling-adapter-instances`. |
+| *gateway_ids*  | yes       | *application-properties* | *string*  | A JSON array containing the IDs of the gateways that may act on behalf of the given device. |
+
+The body of the message SHOULD be empty and will be ignored if it is not.
+
+**Response Message Format**
+
+A response to a *get command-handling adapter instances for device* request contains the [Standard Response Properties]({{< relref "#standard-response-properties" >}}) as well as the properties shown in the following table:
+
+| Name             | Mandatory | Location                 | AMQP Type    | Description |
+| :--------------- | :-------: | :----------------------- | :----------- | :---------- |
+| *content-type*   | no        | *properties*             | *string*     | MUST be set to `application/json` if the invocation of the operation was successful and the body of the response message contains payload as described below. |
+
+The result of a successful invocation is carried in a single Data section of the response message as a UTF-8 encoded string representation of a single JSON object. It is an error to include payload that is not of this type.
+
+The response message JSON object has the following properties:
+
+| Name                | Mandatory | JSON Type     | Description |
+| :------------------ | :-------: | :------------ | :---------- |
+| *adapter-instances* | *yes*     | *array*       | A non-empty array of JSON objects that represent the command-handling adapter instances along with the device or gateway id. |
```

Comment:
wouldn't it be sufficient to just put an array into the response body, i.e. omit the *adapter-instances* property?

Reply:
From a purely API standpoint, yes.
Introducing an JsonArray result would mean that there are different result types for the Device Connection API operations. From an implementation perspective, this means there cannot be just one `DeviceConnectionResult` object used for all methods (extending `RequestResponseResult<JsonObject>`), but there would have to be another `DeviceConnectionArrayResult` (extending `RequestResponseResult<JsonArray>`). 

But, thinking more about it, it is indeed a bit of a crutch, this extra "adapter-instances" field and there probably won't be any other fields, getting added on the same level (like some `last-updated` field for example; wouldn't make sense there).
So, yes, I can change it.

Reply:
I was just wondering. One good reason to keep the JSON obect based result would indeed be to keep it extensible. Maybe we want to add additional properties sometime later? I do see the drawback of having to create another result class so I'll leave it up to you whether you want to change it or not :-)

Reply:
Ok, extensibility and a common `DeviceConnectionResult` - I'll keep it as a JSON object.